ingest/src/DatabaseConnection.ts

```diff
@@ -0,0 +1,114 @@
+import { strict as assert } from "assert";
+import { MongoClient } from "mongodb";
+import { PageStore, PersistedPage } from "./updatePages";
+import { ChunkStore, EmbeddedChunk } from "./updateChunks";
+
+export type DatabaseConnection = {
+  /**
+    Close the connection.
+
+    @param force - Force close, emitting no events
+   */
+  close(force?: boolean): Promise<void>;
+};
+
+/**
+  Create a connection to the database.
+ */
+export const makeDatabaseConnection = async ({
+  connectionUri,
+  databaseName,
+}: {
+  connectionUri: string;
+  databaseName: string;
+}): Promise<DatabaseConnection & PageStore & ChunkStore> => {
```

Comment:
why are these 3 interfaces (`DatabaseConnection`, `PageStore`, and `ChunkStore`) together in the same function? 

could we separate the three, with `PageStore` and `ChunkStore` accepting a `DatabaseConnection` as an argument? it'd provide a greater degree of separation of concern. 

This is the pattern that we use in the Q&A server for the MongoDB interface stuff.

Reply:
I don't see how the -Store interfaces could accept a DatabaseConnection as an argument - I don't want to bake any details about how the "store" works (i.e. database vs. s3 bucket vs. ...) into the interface.

That said, this implementation could separate the actual database connection from the store implementations, but I'm trying to strike the balance here between perfect class implementations and gold plating... I think getting the interfaces right is more important, and it will be easy enough to slice out the database connection implementation later if we want to use it.

Reply:
Note that most of the actual database connection implementation is just using MongoDB APIs so having it separated in the implementation is pretty weak. Trying something...

Reply:
Yeah, it would require me to encode MongoDB APIs (client and db) into the DatabaseConnection interface, which is less great. I'm choosing to stick with this implementation but we can always adapt it later as needed.